### PR TITLE
Allow same-site and same-origin CORS requests

### LINKED ISSUE #1075: Enable 'same origin' CORS settings by default
Actix CORS functionality returns a 400 error if the API Requestor uses and Origin not explicitly allowed for CORS, even if it is a first party or same origin request.
We want to override this behaviour to allow same origin requests as reported by the browser using https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site
Allowing same-origin,same-site, and none

### PR DESCRIPTION
**Pull request by jmbrunskill**
Closes #1075

### CODE REVIEW

**Comment by jmbrunskill** [comment]
Test results
```
jamesbrunskill@Jamess-MBP ~ % curl -v 'http://localhost:8000/graphql' -X 'OPTIONS' -H 'Access-Control-Request-Headers: authorization,content-type' -H 'Access-Control-Request-Method: GET' -H 'Origin: http://example.com' -H'Sec-Fetch-Site: same-site' 2>&1 | grep Origin
> Origin: http://example.com
< vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
jamesbrunskill@Jamess-MBP ~ % curl -v 'http://localhost:8000/graphql' -X 'OPTIONS' -H 'Access-Control-Request-Headers: authorization,content-type' -H 'Access-Control-Request-Method: GET' -H 'Origin: http://example.com' -H'Sec-Fetch-Site: invalid' 2>&1 | grep Origin
> Origin: http://example.com
Origin is not allowed to make this request
jamesbrunskill@Jamess-MBP ~ % curl -v 'http://localhost:8000/graphql' -X 'OPTIONS' -H 'Access-Control-Request-Headers: authorization,content-type' -H 'Access-Control-Request-Method: GET' -H 'Origin: http://example.com' -H'Sec-Fetch-Site: cross-site' 2>&1 | grep Origin
> Origin: http://example.com
Origin is not allowed to make this request
jamesbrunskill@Jamess-MBP ~ % curl -v 'http://localhost:8000/graphql' -X 'OPTIONS' -H 'Access-Control-Request-Headers: authorization,content-type' -H 'Access-Control-Request-Method: GET' -H 'Origin: http://example.com' -H'Sec-Fetch-Site: same-origin' 2>&1 | grep Origin
> Origin: http://example.com
< vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
jamesbrunskill@Jamess-MBP ~ % curl -v 'http://localhost:8000/graphql' -X 'OPTIONS' -H 'Access-Control-Request-Headers: authorization,content-type' -H 'Access-Control-Request-Method: GET' -H 'Origin: http://example.com' -H'Sec-Fetch-Site: none' 2>&1 | grep Origin
> Origin: http://example.com
< vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
jamesbrunskill@Jamess-MBP ~ % curl -v 'http://localhost:8000/graphql' -X 'OPTIONS' -H 'Access-Control-Request-Headers: authorization,content-type' -H 'Access-Control-Request-Method: GET' -H'Sec-Fetch-Site: same-origin' 2>&1 | grep Origin 
Request header `Origin` is required but was not provided
jamesbrunskill@Jamess-MBP ~ % curl -v 'http://localhost:8000/graphql' -X 'GET' -H 'Access-Control-Request-Headers: authorization,content-type' -H 'Access-Control-Request-Method: GET' -H'Sec-Fetch-Site: same-origin' 2>&1 | grep Origin
< vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
jamesbrunskill@Jamess-MBP ~ % curl -v 'http://localhost:8000/graphql' -X 'POST' -H 'Access-Control-Request-Headers: authorization,content-type' -H 'Access-Control-Request-Method: GET' -H'Sec-Fetch-Site: same-origin' 2>&1 | grep Origin
< vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
```
Note: It seems like the 'must include origin' behaviour is only enforced on the OPTIONS requests. GET and POST allow requests without an origin header.

File: server/src/cors.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+use crate::settings::Settings;
+use actix_cors::Cors;
+use actix_web::http::header;
+
+pub fn cors_policy(config_settings: &Settings) -> Cors {
+    let cors = if config_settings.server.develop && config_settings.server.debug_cors_permissive {
+        Cors::permissive()
+    } else {
+        let mut cors_tmp = Cors::default()
```
Comment (clemens-msupply) [inline server/src/cors.rs:9]: Think just `cors` would work here as well (no doing it elsewhere in the code..)

**Review by clemens-msupply — APPROVED** [review]
Thanks looks good

File: server/src/cors.rs (hunk 1 above)
Comment (jmbrunskill) [reply server/src/cors.rs:9]: Rather than cors_tmp? I did wonder about that. Not a fan of shadowing but times like this do make sense...

Comment (clemens-msupply) [reply server/src/cors.rs]: yes shadowing is quite common in Rust and started doing it as well, e.g. in match clauses where you just do type narrowing:

let var = match var {
   Some(var) => var
   None => return
}

but agree shouldn't really be used

**Pull request merged**